lib/components/connection_status.dart

```diff
@@ -0,0 +1,41 @@
+import 'package:firebase_database/firebase_database.dart';
+import 'package:flutter/material.dart';
+import 'package:rtchat/components/chat_panel.dart';
+
+Stream<DateTime?> lastDisconnectTime() {
+  return FirebaseDatabase.instance.ref('.info/connected').onValue.map((event) {
```

Comment:
Does the connection to Firebase DB gets reestablished? Since we don't have any client to that, or do we?

Reply:
Tried it out and the widget seems to leave for a bit and then come back forever :(

Reply:
Ok I added a sentinel watcher, it's a bit hacky but seems to be more reliable...